epochStart/metachain/sovereignEconomics.go

```diff
@@ -0,0 +1,37 @@
+package metachain
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+type sovereignEconomics struct {
+	*economics
+}
+
+// NewSovereignEconomics creates a new sovereign end of epoch economics data object creator
+func NewSovereignEconomics(ec *economics) (*sovereignEconomics, error) {
+	if check.IfNil(ec) {
+		return nil, process.ErrNilEconomicsData
+	}
+
+	ec.baseEconomicsHandler = &sovereignBaseEconomics{
```

Comment:
All nil checks are already inside `ec *` creation (`NewEndOfEpochEconomicsDataCreator(args ArgsNewEpochEconomics)`). That pointer can't be != nil and have nil internal structures.

Do you think it would be worth adding checks anyway?